src/components/DatePicker/CalendarHeader/__tests__/CalendarHeader.spec.tsx

```diff
@@ -22,42 +25,64 @@ describe('CalendarHeader', () => {
     });
 
     it('should render correctly', () => {
-        expect(toJson(wrapper)).toMatchSnapshot();
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('presentation')).toBeInTheDocument();
+        expect(screen.getAllByRole('option')).toHaveLength(YEARS[1] - YEARS[0] + 1);
     });
+
     it('should log an error and render nothing if min year is larger then max year', () => {
         // don't log to test output
         const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementationOnce(() => {});
-        wrapper.setProps({ yearsRange: [9, 2] });
-        wrapper.update();
+        view.rerender(
+            // @ts-ignore
+            <CalendarHeader
+                yearsRange={[9, 2]}
+                date={new Date('2021-12-31')}
+                decreaseMonth={decreaseMonth}
+                increaseMonth={increaseMonth}
+                changeYear={changeYear}
+            />
+        );
 
-        expect(wrapper.html()).toBe(null);
         expect(consoleErrorSpy).toHaveBeenCalledTimes(1);
         expect(consoleErrorSpy.mock.calls[0][0].replace(/\s/g, '')).toMatch(
             `CalendarHeader component has received invalid props.
         Minimum selectable year (9) is larger then maximum selectable year (2)`.replace(/\s/g, '')
         );
     });
+
     it('should fill in the years correctly', () => {
-        const options = wrapper.find('option');
+        const options = screen.getAllByRole('option');
         // eslint-disable-next-line no-plusplus
```

Comment:
If we have in place the rule to not use ++ lets use the other way. If we should use the plus plus, let's change the rules in a follow up ticket to remove this comment.

Reply:
I've create a ticket for that. Could you please check it https://textkernel.atlassian.net/browse/ONEUI-402

Reply:
If it is a ticket just to remove this comment and change the line below to not use ++ I think we don't need a ticket and just do it here. I was more referring if we should look at the eslint rules file regarding this rule (which we might not need to do).

Reply:
Do you mean to add to ?
![image](https://user-images.githubusercontent.com/104063210/236186667-9dcdf001-7e26-44c5-9691-debcac4b4d34.png)

Reply:
I've replaced it to year += 1